File: app/schemas/metrics.py
from pydantic import BaseModel
from typing import Optional


class ApiMetrics(BaseModel):
    api_id: int
    total_checks: int
    successful_checks: int
    failed_checks: int
    uptime_percentage: float
    error_rate: float
    average_response_time: Optional[float] = None
    min_response_time: Optional[float] = None
    max_response_time: Optional[float] = None


class ApiStatus(BaseModel):
    api_id: int
    active: bool
    last_check_success: Optional[bool] = None
    last_checked_at: Optional[str] = None
    current_status: str  # "HEALTHY" | "DOWN" | "UNKNOWN"

File: frontend/src/api/metrics.ts
import client from "./client";

export interface Metrics {
  api_id: number;
  total_checks: number;
  successful_checks: number;
  failed_checks: number;
  uptime_percentage: number;
  error_rate: number;
  average_response_time: number | null;
  min_response_time: number | null;
  max_response_time: number | null;
}

export interface Check {
  id: number;
  status_code: number | null;
  response_time: number | null;
  success: boolean;
  checked_at: string;
}

export const getMetrics = (id: number) =>
  client.get<Metrics>(`/api/monitored-apis/${id}/metrics`).then((r) => r.data);

export const getChecks = (id: number) =>
  client.get<Check[]>(`/api/monitored-apis/${id}/checks`).then((r) => r.data);